gitops/snapshot_test.go

```diff
@@ -113,6 +113,12 @@ var _ = Describe("Gitops functions for managing Snapshots", Ordered, func() {
 		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
 	})
 
+	It("ensures the a decision can be made to NOT promote when the snaphot has not been marked as passed/failed", func() {
+		canBePromoted, reasons := gitops.CanSnapshotBePromoted(hasSnapshot)
+		Expect(canBePromoted).To(BeFalse())
+		Expect(reasons).To(HaveLen(1))
```

Comment:
Maybe we should also explicitly check expected string here

Reply:
Thanks for the suggestion. Added.